presto-main/src/main/java/io/prestosql/sql/planner/optimizations/LimitPushDown.java

```diff
@@ -168,16 +168,24 @@ public PlanNode visitProject(ProjectNode node, RewriteContext<LimitContext> cont
         public PlanNode visitTopN(TopNNode node, RewriteContext<LimitContext> context)
         {
             LimitContext limit = context.get();
+            long count = node.getCount();
+
+            if (limit != null) {
+                count = Math.min(count, limit.getCount());
+            }
+
+            // return empty ValuesNode in case of limit 0
+            if (count == 0) {
+                return new ValuesNode(idAllocator.getNextId(),
```

Comment:
The arguments can all go on the same line. The line's not long enough to warrant splitting it across multiple lines.